### ISSUE TITLE
Maxunpool seems to give a weird error message

### ISSUE THREAD

**Issue by yashdeo55** [pytorch/pytorch#38640]
I am trying to design an autoencoder and when i try to train it i get the error :- **invalid output_size "torch.Size([100, 100])" (dim 0 must be between 96 and 100)** This error shows up at the unpool2 layer . My confusion is that I'st the dim0 already between 96 and 100 ? 
The code is given below :- 
`class T2Encoder2D2(nn.Module):
    def __init__(self):
      """Load the pretrained ResNet-152 and replace top fc layer."""
      super(T2Encoder2D2, self).__init__()
      self.encoder1 = nn.Sequential(nn.Conv2d(1, 16 , kernel_size=(2,2) ,padding = 1),
          nn.ReLU(),
          nn.BatchNorm2d(16),
          nn.MaxPool2d(kernel_size=(2,2) , stride=(2,2), return_indices=True))
      self.encoder2 = nn.Sequential(      
          nn.Conv2d(16, 32 , kernel_size=(2,2),padding = 1),
          nn.ReLU(),
          nn.BatchNorm2d(32),
          nn.MaxPool2d(kernel_size=(2,2) , stride=(2,2), return_indices=True))
      self.encoder3 = nn.Sequential(          
          nn.Conv2d(32, 96, kernel_size=(2,2),padding = 1),
          nn.ReLU(),
          nn.BatchNorm2d(96),
          nn.MaxPool2d(kernel_size=(2,2) , stride=(2,2), return_indices=True))
      self.unpool1 = nn.MaxUnpool2d(kernel_size=(2,2) , stride=(2,2))
      self.decoder1 = nn.Sequential(
          nn.ConvTranspose2d(96, 32 , kernel_size=(2,2),padding = 1),
          nn.ReLU(),
          nn.BatchNorm2d(32))
      self.unpool2 = nn.MaxUnpool2d(kernel_size=(2,2) , stride=(2,2))
      self.decoder2 = nn.Sequential(
          nn.ConvTranspose2d(32, 16, kernel_size=(2,2),padding = 1),
          nn.ReLU(),
          nn.BatchNorm2d(16))
      self.unpool3 = nn.MaxUnpool2d(kernel_size=(2,2) , stride=(2,2))
      self.decoder3 = nn.Sequential(      
          nn.ConvTranspose2d(16, 1, kernel_size=(2,2),padding = 1),
          nn.ReLU(),
          nn.BatchNorm2d(1)
          )
      #self.fc1 = nn.Linear(3960133, 40000)
      #self.rl1 = nn.ReLU()
      #self.fc2 = nn.Linear(10000, 20000)
      #self.rl2 = nn.ReLU()
      #self.fc3 = nn.Linear(10000, 40000)
                
    def forward(self, x):
      size1 = x.size()
      out,ind1 = self.encoder1(x)
      size2 = out.size()
      out,ind2 = self.encoder2(out)
      size3 = out.size()
      out,ind3 = self.encoder3(out)           
      out = self.unpool1(out , ind3 , output_size = size3)      
      out = self.decoder1(out)
      out = self.unpool2(out , ind2 , output_size = size2)      
      out = self.decoder2(out)
      out = self.unpool3(out , ind1 , output_size = size1)      
      out = self.decoder3(out)            
      #out = out.view(out.size(0), -1)     
      #out = self.fc1(out)
      #out = self.rl1(out)
      #out = self.fc2(out)
      #out = self.rl2(out)
      #out = self.fc3(out)
      #out = out.view(1,1,200,200)

      return out  `

**Comment by agolynski** [comment]
@yashdeo55L: I don't know specifics on unpooling implmentation, but the error seems to originate from here:
https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L653

seems like your size need to be between 96 and 100 not inclusively.
